lang/cs/Org.Apache.REEF.Client/API/IREEFClient.cs

```diff
@@ -50,5 +51,12 @@ public interface IREEFClient
         /// <returns></returns>
         [Unstable("0.14", "Working in progress for rest API status returned")]
         Task<FinalState> GetJobFinalStatus(string appId);
+
+        /// <summary>
+        /// Returns all the application reports running in the cluster
+        /// </summary>
+        /// <returns></returns>
+        [Unstable("0.16", "Working in progress for rest API id returned")]
+        Task<List<IApplicationReport>> GetApplicationReports();
```

Comment:
The challenge with this approach is that we'd have to support that API for some time. As a current release of REEF is imminent, you won't be able to replace the current approach with the cleaned-up version in time for a release. Hence, I'd strongly prefer to add `IREEFClientYARN` or such. 

@jwang98052, WDYT?